2023/YoctoSummit-2023.11/fdo/fido-device-onboarding.md

```diff
@@ -0,0 +1,516 @@
+export PKG_CONFIG_PATH="${RECIPE_SYSROOT_NATIVE}/usr/lib/pkgconfig:${RECIPE_SYSROOT_NATIVE}/usr/share/pkgconfig"
+export PKG_CONFIG_LIBDIR="${RECIPE_SYSROOT_NATIVE}/usr/lib/pkgconfig"
+export PKG_CONFIG_DIR="${RECIPE_SYSROOT_NATIVE}/usr/lib/pkgconfig"
+
+# Those variables are handled internally by pkg-config crate.
+# All paths are relative to sysroot, so set PKG_CONFIG_SYSROOT_DIR
+export PKG_CONFIG_SYSROOT_DIR_${TARGET_SYS}="${RECIPE_SYSROOT}"
+export PKG_CONFIG_PATH_${TARGET_SYS}="${RECIPE_SYSROOT}/usr/lib/pkgconfig:${RECIPE_SYSROOT}/usr/share/pkgconfig"
+export PKG_CONFIG_LIBDIR_${TARGET_SYS}="${RECIPE_SYSROOT}/usr/lib/pkgconfig"
+export PKG_CONFIG_DIR_${TARGET_SYS}="${RECIPE_SYSROOT}/usr/lib/pkgconfig"
+```
+
+---
+
+# Bitbake environment vs pkg-config
+
+### openssl-kdf
+
+```rust
+let kdf_h_cts = std::fs::read_to_string("/usr/include/openssl/kdf.h").unwrap();
+```
+
+???
+
+- Cargo's are not standardized
+    + In the above example the library paths are hard-coded
+    + This doesn't work well with Yocto and Bitbake
+
+---
+
+# Bitbake environment vs pkg-config
+
+```rust
+fn read_header(lib: &pkg_config::Library, path_rel: &str) -> std::io::Result<String> {
+    for dir in lib
+        .include_paths
+        .iter()
+        .map(|p| p.as_path())
+        .chain(std::iter::once(std::path::Path::new("/usr/include")))
+    {
+        match std::fs::read_to_string(dir.join(path_rel)) {
+            Ok(r) => return Ok(r),
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => continue,
+            Err(e) => return Err(e),
+        }
+    }
+
+    return Err(std::io::ErrorKind::NotFound.into());
+}
+
+```
+
+```diff
+-   let openssl_version = openssl.version;
++   let openssl_version = &openssl.version;
+```
+
+```diff
+-   let kdf_h_cts = std::fs::read_to_string("/usr/include/openssl/kdf.h").unwrap();
++   let kdf_h_cts = read_header(&openssl, "openssl/kdf.h").unwrap();
+```
+
+???
+
+- Function that uses pkg-config to dynamically find needed libraries
+- `pkg-config` proved most useful despite previously getting in our way
+    + `pkg-config` provides a unified interface for querying installed
+    libraries for the presence of particular software packages, along with
+    their version numbers and other information
+
+---
+
+# Bitbake environment vs pkg-config
+
+### devicemapper-sys
+
+```diff
++   let library = pkg_config::probe_library("devmapper").unwrap();
+```
+
+```diff
++   .clang_args(
++       library.include_paths
++           .iter()
++           .map(|path| format!("-I{}", path.to_string_lossy())),
++   )
+```
+
+???
+
+- Similar story, missing clang arguments
+- pkg-config can be very versatile with its use cases
+
+---
+
+<br>
+<br>
+<br>
+
+## .center[Live Demo]
+
+---
+
+# Resources
+
+- https://fidoalliance.org/
+- https://fido-device-onboard.github.io/docs-fidoiot/latest/
+- https://www.lfedge.org/projects/fidodeviceonboard/
+- https://docs.rs/pkg-config/latest/pkg_config/
+
+---
+
+# Contact us
```

Comment:
I would eliminate this slide; it should also be removed from templates. We have to have a better call to action here.

If we can't put the product in the shop before the event, we should at least mention the fact that if someone would like to talk about FDO integration or would like to hire us for such purposes, one can reach us at some email e.g., leads@3mdeb.com

Reply:
[Added contact to leads@3mdeb.com as well as to me for further questions](https://github.com/3mdeb/conferences/blob/yocto_summit_2023_fdo/2023/YoctoSummit-2023.11/fdo/fido-device-onboarding.md?plain=1#L498-L511)

Reply:
some additional sentence like, if you looking for FIDO related support in Yocto... let there be clear call to action

Reply:
[Addresed](https://github.com/3mdeb/conferences/pull/14/commits/017511c7367804e81dfe8f7a04ba0d03735cebc9)